### PR TITLE
participant-integration-api: Reuse the services execution context for data munging.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
This is the obvious follow-up to #7945. We can share the work-stealing execution pool between the indexer and the API server when they run in the same process.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by SamirTalwar-DA** [comment]
In testing on proprietary code, this seems to get about a 5% performance improvement, which I'm putting down to less CPU contention.

**Review by tudor-da — APPROVED** [review]

**Review by matthiasS-da — APPROVED** [review]
Makes sense. Thank you!

**Review by mziolekda — APPROVED** [review]
Looks good

File: ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala [hunk 1]
```diff
@@ -40,18 +41,21 @@ object JdbcIndexer {
...
         config,
         readService,
+        servicesExecutionContext,
         metrics,
         ExecuteUpdate.owner,
         JdbcLedgerDao.writeOwner(
           serverRole,
           config.jdbcUrl,
           config.eventsPageSize,
+          servicesExecutionContext,
```
Comment (mziolekda) [inline ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala:58]: Samir, this is the change

Comment (SamirTalwar-DA) [reply ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala:58]: This makes total sense. Thanks!

**Pull request merged**